realm/src/main/java/io/realm/dynamic/DynamicRealmObject.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.dynamic;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.RealmObject;
+import io.realm.internal.ColumnType;
+import io.realm.internal.Row;
+import io.realm.internal.Table;
+import io.realm.internal.TableOrView;
+
+/**
+ * Object for interacting with a RealmObject using dynamic names.
+ *
+ * @see io.realm.RealmMigration
+ */
+public class DynamicRealmObject {
+
+     Realm realm;
+     Row row;
+
+    /**
+     * Creates a dynamic Realm object based on a row entry.
+     */
+    public DynamicRealmObject(Realm realm, Row row) {
+        this.realm = realm;
+        this.row = row;
+    }
+
+    public boolean getBoolean(String fieldName) {
+        long columnIndex = row.getColumnIndex(fieldName);
+        checkFieldExists(columnIndex, fieldName);
```

Comment:
No, row is non-checking as it up until now has been hidden behind the typed API where it couldn't fail. So error checking should be done here avoid adding unnecessary overhead to the typed API. Checks could probably be merged though.